File: atcorder/abc125/d/main.py
import sys
sys.setrecursionlimit(10**7)
input = sys.stdin.readline

n = int(input())
a = list(map(int, input().split()))

cnt = 0
mi = 10**9
# -の偶奇と最小値を求める
for i in range(n):
    if a[i] < 0 or a[i] == 0:
        cnt += 1
    mi = min(mi, abs(a[i]))

# 合計
sum = 0
for i in a:
    sum += abs(i)

# 奇数だったら最小値を引く
if cnt % 2 != 0:
    print(sum - mi * 2)
else:
    print(sum)

File: atcorder/abc125/d/main.cpp
#include <bits/stdc++.h>
#include <numeric>
using namespace std;

#define ALL(v) v.begin(), v.end()
#define V vector
#define P pair
#define ld long double
#define ll long long

const int MOD =  1000000007;
const int IINF = INT_MAX;
const ll INF = 1LL << 60;


int main() {
    int n; cin >> n;
    int cnt = 0;
    ll sum = 0;
    ll mi = INF;
    for(int i = 0; i < n; i++){
        ll a; cin >> a;
        mi = min(abs(a), mi);
        sum += abs(a);
        if(a < 0) cnt++;
    }

    if(cnt & 1) cout << sum - mi * 2 << endl;
    else cout << sum << endl;

    return 0;
}
